File: 프로그래머스/lv2/84512. 모음 사전/모음 사전.py
import time
def solution(word):
    alphabets = ['A', 'E', 'I', 'O', 'U']
    global answer
    answer = 0
    
    def dfs(string):
        global answer
        answer += 1
        if string == word:
            return True
        
        if len(string) == 5:
            return False
        
        for alphabet in alphabets:
            if dfs(string + alphabet):
                return True
        
    if dfs(""):
        return answer - 1

File: 프로그래머스/2/84512. 모음 사전/모음 사전.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;
string dict = "AEIOU";
int count = 0;
int answer = 0;
bool found = false;

void dfs(int idx, string s, string word) {
    if (found) return;
    count++;
    
    if (word == s) {
        answer = count;
        found = true;
        return;
    }
    
    if (idx == 5) {
        return;
    }
    
    for (int i = 0; i < dict.size(); i++) {
        dfs(idx + 1, s + dict[i], word);
    }
}

int solution(string word) {
    for (int i = 0; i < dict.size(); i++) {
        dfs(1, string(1, dict[i]), word);
        if (found) break;
    }
    
    return answer;
}